### DISCUSSION TITLE
Long Indicies

### DISCUSSION THREAD

**Discussion by marc-ernst** [eclipse-nattable/nattable#135] [Ideas]
We have a huge data set. Even if we have lazy loading implemented we should have the indicies in NatTable supporting Long, otherwise it's complicated to handle the data.

**Comment by fipro78** [comment]
Hi,

This request was raised some years ago the last time. With no further interest after some discussions.

Changing the indexes from int to long means:

- a major API change which affects almost every class in the project (layers, commands, events, handlers, etc)
- it affects all downstream projects and consumers which would need to change their code

Having in mind that such a change would cause days, maybe even weeks of development, I doubt that such a change will happen NatTable.

Also note that NatTable operates on data in memory, which means multiple GBs of RAM would be necessary to be able to access the data in NatTable.
